engine/schema/src/org/apache/cloudstack/resourcedetail/ResourceDetailsDaoBase.java

```diff
@@ -104,6 +104,15 @@ public void removeDetail(long resourceId, String key) {
         }
     }
 
+    public void removeDetails(String key) {
```

Comment:
this is needed because existing method works for input params resourceId, Name . To make the solution specific to volumes moved this method to VolumeDetailsDao and implemented